react/src/components/molecules/AccordionItem/index.js

```diff
@@ -16,54 +16,61 @@ class AccordionItem extends React.Component {
   }
 
   handleClick() {
+    const { open } = this.state;
     this.setState({
-      open: !this.state.open
+      open: !open
     });
   }
 
   render() {
+    const {
+      buttonId, secondary, border, emphasize, icon, title, headerLevel, children, expandableSectionId, info
+    } = this.props;
+    const { open } = this.state;
     const accordionClasses = classNames({
-      'ma__accordion-item': !this.props.secondary,
-      'ma__accordion-item--secondary': this.props.secondary,
-      'ma__accordion-item--borderless': !this.props.border && !this.props.secondary
+      'ma__accordion-item': !secondary,
+      'ma__accordion-item--secondary': secondary,
+      'ma__accordion-item--borderless': !border && !secondary
     });
     const buttonClasses = classNames({
-      'ma__accordion-header__button': !this.props.secondary,
-      'ma__accordion-header__button--secondary': this.props.secondary,
-      'is-open': this.state.open,
-      'ma__accordion-header__button--solid': this.props.emphasize && !this.props.secondary,
-      'ma__accordion-header__button--trans': !this.props.emphasize && !this.props.secondary
+      'ma__accordion-header__button': !secondary,
+      'ma__accordion-header__button--secondary': secondary,
+      'is-open': open,
+      'ma__accordion-header__button--solid': emphasize && !secondary,
+      'ma__accordion-header__button--trans': !emphasize && !secondary
     });
     const headingClasses = classNames({
-      'ma__accordion-header__title': !this.props.secondary,
-      'ma__accordion-header__title--secondary': this.props.secondary
+      'ma__accordion-header__title': !secondary,
+      'ma__accordion-header__title--secondary': secondary
     });
 
     return(
       <div className={accordionClasses}>
         <header className="ma__accordion-header">
           <button
             className={buttonClasses}
-            aria-label={this.props.info}
+            aria-label={info}
             onClick={this.handleClick}
-            aria-expanded={this.state.open}
+            aria-expanded={open}
+            aria-controls={expandableSectionId}
+            id={buttonId}
           >
-            { this.props.icon && !this.props.secondary && (
+            { icon && !secondary && (
               <div className="ma__accordion-header__icon">
-                {this.props.icon}
+                {icon}
               </div>
             )}
-            { this.props.secondary && (
+            { secondary && (
               <div className="ma__accordion-header__icon--secondary">
                 <Icon name="chevron" svgHeight={20} svgWidth={20} />
               </div>
             )}
-            <Heading class={headingClasses} text={this.props.title} level={this.props.headerLevel} />
+            <Heading class={headingClasses} text={title} level={headerLevel} />
```

Comment:
I wonder if the `aria-labelledby` id should be on the heading instead of the button which contains the label text

Reply:
i haven't found an example that the referenced id is on the outter div with aria-label

Reply:
based on w3 --> https://www.w3.org/TR/wai-aria-practices/examples/accordion/accordion.html it should be on the button level event if nested title